### PR TITLE
Change docstring to Google format

### PR DESCRIPTION
**Pull request by Narann**
As requested in https://github.com/sqlboy/fileseq/issues/68, docstring format is now Google one.

The current documentation config rely on restructured text. So the generated doc is not that nice. What can I do in this regard?

Unit tests pass except the ones using what seems to be an absolute path.

### CODE REVIEW

File: src/fileseq/filesequence.py [hunk 1]
```diff
@@ -198,36 +212,42 @@ def frameSet(self):
...
         Set a new file extension for the sequence.
 
-        .. note::
+        Note:
             A leading period will be added if none is provided.
 
-        :param ext: the new file extension
-        :rtype: None
+        Args:
+            ext (str):
```
Comment (justinfx) [inline src/fileseq/filesequence.py]: Forgot to preserve the description text for the "ext" field

File: src/fileseq/filesequence.py [hunk 2]
```diff
@@ -198,36 +212,42 @@ def frameSet(self):
...
+
+        Returns:
+            None:
```
Comment (justinfx) [inline src/fileseq/filesequence.py]: This isn't needed if nothing is being returned

File: src/fileseq/filesequence.py [hunk 3]
```diff
@@ -257,8 +281,9 @@ def setFrameRange(self, frange):
         """
         Set a new frame range for the sequence.
 
-        :param frange: a properly formatted frame range, as per :class:`fileseq.frameset.FrameSet`
-        :rtype: None
+        Args:
+            frange: a properly formatted frame range, as per
```
Comment (justinfx) [inline src/fileseq/filesequence.py]: Missing type:  `frange (str)`

File: src/fileseq/frameset.py [hunk 4]
```diff
@@ -74,17 +67,31 @@ def __new__(cls, *args, **kwargs):
         """
         Initialize the :class:`FrameSet` object.
 
-        :type frange: str
-        :param frange: the frame range as a string (ie "1-100x5")
-        :returns: the :class:`FrameSet` instance
-        :raises: :class:`fileseq.exceptions.ParseException` if the frame range
-                 (or a portion of it) could not be parsed
+        Args:
+            frange (str or FrameSet): the frame range as a string (ie "1-100x5")
```
Comment (justinfx) [inline src/fileseq/frameset.py]: Missing rst formatting:  
```
:class:`FrameSet`
```

File: src/fileseq/frameset.py [hunk 5]
```diff
@@ -230,9 +244,15 @@ def _cast_to_frameset(cls, other):
         """
         Private method to simplify comparison operations.
 
-        :param other: the :class:`FrameSet`, set, frozenset, or iterable to be compared
-        :rtype: :class:`FrameSet`
-        :returns: :class:`NotImplemented` if a comparison is impossible
+        Args:
+            other: the :class:`FrameSet`, set, frozenset, or iterable to be
```
Comment (justinfx) [inline src/fileseq/frameset.py]: Should be:

```
other (:class:`FrameSet` or set or frozenset or or iterable): item to be compared 
```

File: src/fileseq/filesequence.py [hunk 6]
```diff
@@ -198,36 +212,42 @@ def frameSet(self):
...
 
     def setFrameSet(self, frameSet):
         """
         Set a new :class:`fileseq.frameset.FrameSet` for the sequence.
 
-        :param frameSet: the new :class:`fileseq.frameset.FrameSet` object
-        :rtype: None
+        Args:
+            frameSet (:class:`fileseq.frameset.FrameSet`): the new
+            :class:`fileseq.frameset.FrameSet` object
```
Comment (justinfx) [inline src/fileseq/filesequence.py]: Sphinx complains about this missing a newline after the last Args field

**Comment by justinfx** [comment]
For the sphinx-build to process the google docstrings correctly you need to update the docs/conf.py:

```
extensions = [
    'sphinx.ext.autodoc',
    'sphinx.ext.viewcode',
    'sphinx.ext.napoleon',  # <---
]
```

Also, I would recommend not moving the class level documentation into the `__init__` methods. Sphinx will not document them by default unless you set `napoleon_include_init_with_doc = True`. But even then, the methods are all sorted so the constructor docs get shown much further down. Its cleaner to just have them all in the class docstring.

**Comment by justinfx** [comment]
Also I forgot to say.... Thanks for this work!

**Comment by Narann** [comment]
I've fixed all what you pointed, moved the init doc string to the class one.

I've also update some domains to have proper link in generated doc. I'm sure there must be a way to avoid full namespaces and have `FrameSet` instead of `fileseq.frameset.FrameSet` but I didn't find how.

Hope it's OK. :)

**Comment by justinfx** [comment]
Thanks for the fixes.
The way to do relative references would be `.FrameSet`

**Comment by Narann** [comment]
> The way to do relative references would be `.FrameSet`

Perfect, let me fix this ! :)

**Comment by Narann** [comment]
Dopc looks nicer now.

I realize there is some docstring calling an unknown `FrameRange` class.

**Comment by justinfx** [comment]
I fixed up those old docstring refs. Thanks again!

**Pull request merged**